pyccel/codegen/printing/ccode.py

```diff
@@ -1033,15 +1036,21 @@ def _print_Constant(self, expr):
         return self._print(val)
 
     def _print_Return(self, expr):
-        code = ''
         args = [VariableAddress(a) if self.stored_in_c_pointer(a) else a for a in expr.expr]
-        if expr.stmt:
-            code += self._print(expr.stmt)+'\n'
+        if len(args) > 1:
+            if expr.stmt:
+                return self._print(expr.stmt)+'\n'+'return 0;'
+            return 'return 0;'
         if len(args) == 1:
-            code +='return {0};'.format(self._print(args[0]))
-        elif len(args) > 1:
-            code += 'return 0;'
-        return code
+            if isinstance(expr.stmt, CodeBlock):
+                for a in expr.stmt.body:
+                    if isinstance(a, Assign):
+                        b = a
```

Comment:
Do we definitely need this loop? Do we want the last `Assign`? Or the last element and printing the assignment can be avoided if the last element is an `Assign`?

Reply:
@EmilyBourne  Yes we do need this loop as all `Deallocate` statements are put in `Return.stmt` and not all (or necessarily any of them) have anything to do with the value which is returned.

@mostafaMamoni If I ask a question it is because I am not certain of the answer. In this case your code above was correct. If you can justify why something is written like that then this is usually a better answer than just changing it